### PR TITLE
[feat] --check-lora-weight-equal LoRA adapter weight-sync checker

### PR DESCRIPTION
**Pull request by Zhichenzzz**
## Summary
- Adds `--check-lora-weight-equal`, the LoRA analogue of the existing `--check-weight-update-equal` (base weights only).
- On every megatron->sglang LoRA sync, the trainer computes a per-tensor sha256 manifest of the adapter it's about to send and passes it as `expected_checksums` alongside the serialized tensors.
- Each rollout engine hashes the tensors it actually received after deserializing and asserts they match, raising on any mismatch/missing name.
- Sglang-side verification (`expected_checksums` handling in `tp_worker.py`/`lora_manager.py`) already exists in the sglang-miles fork https://github.com/sgl-project/sglang/pull/31831

### CODE REVIEW

File: miles/backends/megatron_utils/update_weight/update_weight_from_tensor.py [hunk 1]
```diff
@@ -363,6 +365,17 @@ def _send_to_colocated_engine(
...
+            expected_checksums = None
+            if check_equal:
+                import hashlib
+
+                expected_checksums = {
+                    n: hashlib.sha256(
+                        t.detach().cpu().contiguous().flatten().view(torch.uint8).numpy().tobytes()
+                    ).hexdigest()
+                    for n, t in hf_named_tensors
+                }
```
[the remark this answers is not in the document]
Comment (Zhichenzzz) [reply miles/backends/megatron_utils/update_weight/update_weight_from_tensor.py:376]: not correct

File: miles/utils/arguments.py [hunk 2]
```diff
@@ -1717,6 +1717,18 @@ def add_debug_arguments(parser):
...
+            parser.add_argument(
+                "--check-lora-weight-equal",
+                action="store_true",
+                default=False,
+                help=(
+                    "Verify the megatron->sglang LoRA adapter weight-sync: on every sync the "
+                    "trainer ships a per-tensor sha256 manifest of the adapter it sends, and "
+                    "each rollout engine hashes the tensors it received and asserts they match "
+                    "(raising on any mismatch/missing name). The LoRA analogue of "
+                    "--check-weight-update-equal, which only covers base weights."
```
Comment (yushengsu-thu) [inline miles/utils/arguments.py]: The flag only covers the colocated `from_tensors` path — the distributed sync (`update_weight_from_distributed/broadcast.py:139` -> `load_lora_adapter_from_distributed`) doesn't carry a manifest, so "on every sync" overclaims. Also "raising" will be wrong once the receiver side is fixed per the review on sgl-project/sglang#31831 (fail the load with `success=False`; `_check_weight_sync_results` already raises trainer-side on that).

```suggestion
                    "Verify the megatron->sglang LoRA adapter weight-sync on the colocated "
                    "(from_tensors) path: on every sync the trainer ships a per-tensor sha256 "
                    "manifest of the adapter it sends, and each rollout engine hashes the "
                    "tensors it received and fails the load on any mismatch/missing/extra "
                    "name. Requires sglang-miles with sgl-project/sglang#31831. The LoRA "
                    "analogue of --check-weight-update-equal, which only covers base weights."
```

**Review by yushengsu-thu — COMMENTED** [review]
Sender side looks right: the hash recipe is byte-for-byte the same as the receiver's (`detach().cpu().contiguous().flatten().view(torch.uint8).numpy().tobytes()`), it's computed once on the gather-src over the full local adapter — which is valid for every rank since the Bridge exports are broadcast/all_gather'd identically, and it doubles as a cross-rank export-divergence check — and there's zero overhead with the flag off.

Two things before this lands though:

1. The description says the sglang-side handling "already exists in the sglang-miles fork" — it doesn't yet. That's exactly sgl-project/sglang#31831 (and it lives in `tp_worker.py`, not `lora_manager.py`). This PR needs to land **after** #31831: if it ships first and someone flips the flag, the server just drops the unknown field (msgspec ignores extras) and the check silently no-ops — worst kind of false confidence. Worth fixing the description too.

2. Help-text wording — suggestion inline (scope to the colocated path, note the #31831 requirement, and "fails the load" instead of "raising" pending the failure-path change requested on #31831).

Code itself LGTM once those are sorted.

**Review by yushengsu-thu — APPROVED** [review]

**Review by yueming-yuan — APPROVED** [review]
